### PR TITLE
Add a public replacement for OC::$server->get

### PR DESCRIPTION
**Pull request by CarlSchwan**
Currently, most app are using OC::$sever->get, a private API to load some services. While apps should be migrating to dependency injections, this is not always possible. This PR adds a nicer public API to load services.

<s>Potentially we should expose IServerContainer too and just have `Server::get` has a nice alias for `Server::getServerContainer()->get`</s>

### CODE REVIEW

**Comment by PVince81** [comment]
related older PR: https://github.com/nextcloud/server/pull/18995

**Comment by come-nc** [comment]
> Potentially we should expose IServerContainer too and just have `Server::get` has a nice alias for `Server::getServerContainer()->get`

Can you detail this? I see no `Server::getServerContainer()->get` in the PR

**Comment by come-nc** [comment]
Otherwise looks clean enough.

**Review by come-nc — APPROVED** [review]

**Comment by PVince81** [comment]
- [ ] add to https://github.com/nextcloud/server/issues/32117 after merging @CarlSchwan

**Comment by CarlSchwan** [comment]
@ChristophWurst I remember you had an opinion on that, so I would appreciate your review on it :)

**Review by ChristophWurst — APPROVED** [review]
With great power comes great responsibility

**Comment by CarlSchwan** [comment]
> Would be lovely to also have the dev docs updated for this

https://github.com/nextcloud/documentation/pull/8294 :) I also updated that page on the way

File: lib/public/Server.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+// SPDX-FileCopyrightText: 2022 Carl Schwan <carl@carlschwan.eu>
```
Comment (skjnldsv) [inline lib/public/Server.php:4]: Is it a new fancy thing 😮

Comment (CarlSchwan) [reply lib/public/Server.php:4]: Yeah it a new fancy stuff, but we are not using it in server so let's me move back to the old way to declare licenses

For context https://github.com/nextcloud/server/issues/29723 and https://github.com/nextcloud/server/issues/18138 and I moved to that in the profiler app with a workflow to check the compliance https://github.com/nextcloud/profiler/blob/master/.github/workflows/reuse.yml

**Comment by CarlSchwan** [comment]
ci failure unrelated

**Comment by skjnldsv** [comment]
> * [ ]  add to [Critical changes for developers & admins for Nextcloud 25 #32117](https://github.com/nextcloud/server/issues/32117) after merging @CarlSchwan

PIng

**Comment by skjnldsv** [comment]
Docs in
- https://github.com/nextcloud/documentation/pull/8294
- https://github.com/nextcloud/documentation/pull/10632

**Pull request merged**